lib/modules/manager/devcontainer/extract.spec.ts

```diff
@@ -0,0 +1,363 @@
+import upath from 'upath';
+import * as dockerfileExtract from '../dockerfile/extract';
+import { extractPackageFile } from '.';
+
+function getFixture(content: string) {
+  return {
+    content,
+    path: upath.resolve(__dirname, 'devcontainer.json'),
+  };
+}
+
+describe('modules/manager/devcontainer/extract', () => {
+  describe('extractPackageFile()', () => {
+    afterEach(() => {
+      jest.restoreAllMocks();
+    });
+
+    it('returns null when the dev container JSON file is empty', () => {
+      // Arrange
+      const content = '';
+      const packageFile = '';
+      const extractConfig = {};
+      // Act
+      const result = extractPackageFile(content, packageFile, extractConfig);
+
+      // Assert
+      expect(result).toBeNull();
+    });
+
+    it('returns null when the dev container JSON file contents are malformed', () => {
+      // Arrange
+      const content = 'malformed json}}}';
+      const packageFile = '';
+      const extractConfig = {};
+      // Act
+      const result = extractPackageFile(content, packageFile, extractConfig);
+
+      // Assert
+      expect(result).toBeNull();
+    });
+
+    it('returns feature image deps when only the features property is defined in dev container JSON file', () => {
+      // Arrange
+      const fixture = getFixture(`
```

Comment:
```suggestion
      const fixture = getFixture(codeBlock`
```

use `codeBlock` template helper